src/Host/Broker/Impl/Sessions/SessionManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Linq;
+using System.Security;
+using System.Security.Principal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.R.Host.Broker.Interpreters;
+using Microsoft.R.Host.Broker.Logging;
+using Microsoft.R.Host.Broker.Pipes;
+
+namespace Microsoft.R.Host.Broker.Sessions {
+    public class SessionManager {
+        private readonly InterpreterManager _interpManager;
+        private readonly LoggingOptions _loggingOptions;
+        private readonly ILogger _hostOutputLogger, _messageLogger;
+
+        private readonly Dictionary<string, List<Session>> _sessions = new Dictionary<string, List<Session>>();
```

Comment:
They aren't, since there's no session delete functionality here yet - that one is TBD.

---
In reply to: [74966786](https://github.com/Microsoft/RTVS/pull/2173#discussion_r74966786) [](ancestors = 74966786)